### PR TITLE
Updated package references to latest versions

### PR DESCRIPTION
**Pull request by aivascu**
When using the library with newer version of NSubstitute I get package conflicts.
Probably due to some breaking changes since version `3.1.0`.

### CODE REVIEW

File: tests/EntityFrameworkMock.NSubstitute.Tests/DbContextMockTests.cs [hunk 1]
```diff
@@ -96,7 +96,6 @@ void CreateDbMock()
         public void DbContextMock_CreateDbSetMock_ShouldSetupMockForDbSetSelector()
         {
             var dbContextMock = new DbContextMock<TestDbContext>("abc");
-            Assert.That(dbContextMock.Object.Users, Is.Null);
```
Comment (huysentruitw) [inline tests/EntityFrameworkMock.NSubstitute.Tests/DbContextMockTests.cs]: What's the reason for removing this assertion?

**Review by huysentruitw — CHANGES_REQUESTED** [review]

File: tests/EntityFrameworkMock.NSubstitute.Tests/DbContextMockTests.cs (hunk 1 above)
Comment (aivascu) [reply tests/EntityFrameworkMock.NSubstitute.Tests/DbContextMockTests.cs]: That assertion was failing. It seems there have been some changes in the way NSubstitute does partial mocking. 
When the `Users` property is invoked the `DbContext` attempts to connect to the data provider using the `"abc"` connection string and then throws an error due to an nonexistent resource.
I thought of setting all the `DbSet`s to null by default when creating the `DbContextMock<>` instance, as a hotfix, to preserve the behavior, but I wanted to consult you first.

Comment (huysentruitw) [reply tests/EntityFrameworkMock.NSubstitute.Tests/DbContextMockTests.cs]: Thanks for your patience. Had to look into this change, the real fix here is to change `Is.Null` with `Is.Empty`, like this:

```
Assert.That(dbContextMock.Object.Users, Is.Empty);
```

Comment (aivascu) [reply tests/EntityFrameworkMock.NSubstitute.Tests/DbContextMockTests.cs]: @huysentruitw tried to change the test as per your suggestion and it does not work (see the test log below).
The `DbContext` is still trying to connect to the database using the connection string.
I think this kinda makes sense for the virtual member to act as the original implementation unless explicitly overriden in the test, especially if we're talking about a partial mock (aka `ForPartsOf<>()` ).
The only way I could obtain the original behavior is by changing the mock type from partial to full, by using `For<>()`.
Check the NSub `DbContextMock.cs` class, to see the change.

Test result using `Is.Empty`:
```
Test Name:	DbContextMock_CreateDbSetMock_ShouldSetupMockForDbSetSelector
Test FullName:	EntityFrameworkMock.NSubstitute.Tests.DbContextMockTests.DbContextMock_CreateDbSetMock_ShouldSetupMockForDbSetSelector
Test Source:	C:\src\entity-framework-mock\tests\EntityFrameworkMock.NSubstitute.Tests\DbContextMockTests.cs : line 96
Test Outcome:	Failed
Test Duration:	0:01:00.029

Result StackTrace:	
at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
   at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at System.Data.Entity.Infrastructure.Interception.DbConnectionDispatcher.<Open>b__36(DbConnection t, DbConnectionInterceptionContext c)
   at System.Data.Entity.Infrastructure.Interception.InternalDispatcher`1.Dispatch[TTarget,TInterceptionContext](TTarget target, Action`2 operation, TInterceptionContext interceptionContext, Action`3 executing, Action`3 executed)
   at System.Data.Entity.Infrastructure.Interception.DbConnectionDispatcher.Open(DbConnection connection, DbInterceptionContext interceptionContext)
   at System.Data.Entity.SqlServer.SqlProviderServices.<>c__DisplayClass33.<UsingConnection>b__32()
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.<>c__DisplayClass1.<Execute>b__0()
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute[TResult](Func`1 operation)
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute(Action operation)
   at System.Data.Entity.SqlServer.SqlProviderServices.UsingConnection(DbConnection sqlConnection, Action`1 act)
   at System.Data.Entity.SqlServer.SqlProviderServices.UsingMasterConnection(DbConnection sqlConnection, Action`1 act)
   at System.Data.Entity.SqlServer.SqlProviderServices.CreateDatabaseFromScript(Nullable`1 commandTimeout, DbConnection sqlConnection, String createDatabaseScript)
   at System.Data.Entity.SqlServer.SqlProviderServices.DbCreateDatabase(DbConnection connection, Nullable`1 commandTimeout, StoreItemCollection storeItemCollection)
   at System.Data.Entity.Core.Common.DbProviderServices.CreateDatabase(DbConnection connection, Nullable`1 commandTimeout, StoreItemCollection storeItemCollection)
   at System.Data.Entity.Core.Objects.ObjectContext.CreateDatabase()
   at System.Data.Entity.Migrations.Utilities.DatabaseCreator.Create(DbConnection connection)
   at System.Data.Entity.Migrations.DbMigrator.EnsureDatabaseExists(Action mustSucceedToKeepDatabase)
   at System.Data.Entity.Migrations.DbMigrator.Update(String targetMigration)
   at System.Data.Entity.Internal.DatabaseCreator.CreateDatabase(InternalContext internalContext, Func`3 createMigrator, ObjectContext objectContext)
   at System.Data.Entity.Internal.InternalContext.CreateDatabase(ObjectContext objectContext, DatabaseExistenceState existenceState)
   at System.Data.Entity.Database.Create(DatabaseExistenceState existenceState)
   at System.Data.Entity.CreateDatabaseIfNotExists`1.InitializeDatabase(TContext context)
   at System.Data.Entity.Internal.InternalContext.<>c__DisplayClassf`1.<CreateInitializationAction>b__e()
   at System.Data.Entity.Internal.InternalContext.PerformInitializationAction(Action action)
   at System.Data.Entity.Internal.InternalContext.PerformDatabaseInitialization()
   at System.Data.Entity.Internal.LazyInternalContext.<InitializeDatabase>b__4(InternalContext c)
   at System.Data.Entity.Internal.RetryAction`1.PerformAction(TInput input)
   at System.Data.Entity.Internal.LazyInternalContext.InitializeDatabaseAction(Action`1 action)
   at System.Data.Entity.Internal.LazyInternalContext.InitializeDatabase()
   at System.Data.Entity.Internal.InternalContext.Initialize()
   at System.Data.Entity.Internal.InternalContext.GetEntitySetAndBaseTypeForType(Type entityType)
   at System.Data.Entity.Internal.Linq.InternalSet`1.Initialize()
   at System.Data.Entity.Internal.Linq.InternalSet`1.GetEnumerator()
   at System.Data.Entity.Infrastructure.DbQuery`1.System.Collections.IEnumerable.GetEnumerator()
   at NUnit.Framework.Constraints.CollectionConstraint.IsEmpty(IEnumerable enumerable)
   at NUnit.Framework.Constraints.EmptyCollectionConstraint.Matches(IEnumerable collection)
   at NUnit.Framework.Constraints.CollectionConstraint.ApplyTo[TActual](TActual actual)
   at NUnit.Framework.Constraints.EmptyConstraint.ApplyTo[TActual](TActual actual)
   at NUnit.Framework.Assert.That[TActual](TActual actual, IResolveConstraint expression, String message, Object[] args)
   at NUnit.Framework.Assert.That[TActual](TActual actual, IResolveConstraint expression)
   at EntityFrameworkMock.NSubstitute.Tests.DbContextMockTests.DbContextMock_CreateDbSetMock_ShouldSetupMockForDbSetSelector() in C:\src\entity-framework-mock\tests\EntityFrameworkMock.NSubstitute.Tests\DbContextMockTests.cs:line 99
Result Message:	
System.Data.SqlClient.SqlException : A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)
Data:
  HelpLink.ProdName: Microsoft SQL Server
  HelpLink.EvtSrc: MSSQLServer
  HelpLink.EvtID: -1
  HelpLink.BaseHelpUrl: http://go.microsoft.com/fwlink
  HelpLink.LinkId: 20476
```

**Comment by huysentruitw** [comment]
Thank you!

**Comment by huysentruitw** [comment]
Packages 1.0.0.57 pushed to NuGet

**Pull request merged**